doc: clarify mkdir() recursive behavior (#48109)
nodejs/node: doc/api/fs.md

The callback is given a possible exception and, if `recursive` is `true`, the
first directory path created, `(err[, path])`.
`path` can still be `undefined` when `recursive` is `true`, if no directory was

> created,for instance if the directory was previously created or other issues.

Reviewer: ```suggestion
created (for instance, if it was previously created).
```

The optional `options` argument can be an integer specifying `mode` (permission
and sticky bits), or an object with a `mode` property and a `recursive`
property indicating whether parent directories should be created. Calling
`fs.mkdir()` when `path` is a directory that exists results in an error only

> when `recursive` is false.That is, if we don't add `recursive` as true and  
> the file was previously created,we get an EEXIST (error Exist) message telling
>  us path already exists.

Reviewer: ```suggestion
when `recursive` is false. If `recursive` is false and  the directory exists,
an `EEXIST` error occurs.
```

```mjs
import { mkdir } from 'node:fs';

> //Note the relative path to `tmp` below ,otherwise  path won't be created in our project.
> 
> mkdir('./tmp/a/apple', { recursive: true }, (err) => {

Reviewer: ```suggestion
// Create ./tmp/a/apple, regardless of whether ./tmp and ./tmp/a exist.
mkdir('./tmp/a/apple', { recursive: true }, (err) => {
```

Reviewer: (The code should work fine with an absolute path instead of a relative path, but there are security implications to creating directories in /tmp with predictable names that aren't accounted for in the example code. It shouldn't ignore if someone else has already created the directory with permissions different than expected or with pre-existing contents. So I think it's good to switch it to a relative directory like you've done here or else some other absolute directory.)

Author: I think I have a better explanation to the whole behaviour of the code above ,can I go ahead and fix it?

  if (err) throw err;
});
```

> without `recursive` being set as true it the code above,output shows an error
> 
> 

Reviewer: ```suggestion
```
